server/server.js

```diff
@@ -83,18 +83,26 @@ app.post("/addContact", async (req, res) => {
 
 app.put("/editContact/:id", async (req, res) => {
   const contactId = req.params.id;
-  const { newName, newEmail, newPhone, newNotes } = req.body;
+  const { name, email, phone, notes } = req.body;
+
   try {
     const result = await pool.query(
       "UPDATE contacts SET name = $1, email = $2, phone = $3, notes = $4 WHERE contact_id = $5 RETURNING contact_id",
-      [newName, newEmail, newPhone, newNotes, contactId]
+      [name, email, phone, notes, contactId]
     );
-    res.status(200).json(result.rows[0], {
+
+    if (result.rowCount === 0) {
+      res.status(404).json({ error: "Contact not found" });
+      return;
+    }
```

Comment:
Added this code to return an error if no contact was updated. The SQL to UPDATE a row in the table will successfully complete even if no row was found or updated. So have to manually check how many rows were updated. If 1 row was updated, that means it was successful. If none, that means it couldn't find any contact to update.